### ISSUE TITLE
feature request: multiple selector/command sets

### ISSUE THREAD

**Issue by bgrayburn** [cytoscape/cytoscape.js-cxtmenu#59]
forgive me if this is already possible, but I haven't been able to find out how to create multiple sets of commands that get show based on different selectors. Presumably it'd be first match, or maybe merge when multiple selectors match. Let me know if this would be a good feature, I'll try to make time to implement this if it is. Would just manage multiple instances.

**Comment by bgrayburn** [comment]
sorry didn't mean to close it

**Comment by maxkfranz** [comment]
`commands` can be `function( node ){ return [ /* ... */ ]; }` so the commands can be dynamic.  You don't have to use the `selector` to differentiate elements: `selector` could be `node, edge` and `commands()` could specify everything.

**Comment by bgrayburn** [comment]
I definitely see the use in that and even how it can be used to accomplish this to some extent. But now if I want all the nodes with a certain class to have a cxtmenu but not the others, would I just return an empty array from the commands function for those without the class? What if I want a single node to have a menu and none others, does the command function need to run for each node? Feel free to close this, I understand how to solve my immediate situation given your advice. Thank you for this and this awesome library! Lots of fun to work with.

**Comment by maxkfranz** [comment]
You can use both the commands function and a selector if you like.  

You can alternatively use `enabled` per command.  This was missed in the readme.  This is usually the best solution, because you maintain the user's muscle memory.

**Comment by bgrayburn** [comment]
Sorry I never got back to you on this thanks for the solution and great
library!

On Oct 23, 2017 11:06 AM, "Max Franz" <notifications@github.com> wrote:

Closed #59 <https://github.com/cytoscape/cytoscape.js-cxtmenu/issues/59>.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
<https://github.com/cytoscape/cytoscape.js-cxtmenu/issues/59#event-1305842090>,
or mute the thread

**Comment by mkhoussid** [comment]
I just want to point out, that the following code works,
```
ctxMenuRef.current = cyRef.current.cxtmenu({
     selector: 'node, edge', // or '*'
     commands: (nodeOrEdge) => {
          if (nodeOrEdge.isNode()) {
               // do something with node
          } else if (nodeOrEdge.isEdge()) {
               // do something with edge
          }
     }
})
```
and this works,
```
ctxMenuRef.current = cyRef.current.cxtmenu({
     selector: 'core',
     commands: (nodeOrEdge) => {
          // do something with core
     }
})
```
but this does not work,
```
ctxMenuRef.current = cyRef.current.cxtmenu({
     selector: 'node, edge, core', // <-- core is never triggered in this case
     commands: (nodeOrEdge) => { . . . }
})
```
